extension.ts

```diff
@@ -77,170 +76,175 @@ function organizeAll(useRegions: boolean, addPublicModifierIfMissing: boolean, a
 
 }
 
-function organize(editor: vscode.TextEditor | undefined, useRegions: boolean, addPublicModifierIfMissing: boolean, addRegionIdentation: boolean, addRegionCaptionToRegionEnd: boolean, groupElementsWithDecorators: boolean)
+function organizeCode(sourceCode: string, fileName: string, useRegions: boolean, addPublicModifierIfMissing: boolean, addRegionIdentation: boolean, addRegionCaptionToRegionEnd: boolean, groupElementsWithDecorators: boolean)
 {
-    let sourceFile: ts.SourceFile;
-    let sourceCode: string;
-    let edit: vscode.WorkspaceEdit;
-    let elements: ElementNode[];
-    let start: vscode.Position;
-    let end: vscode.Position;
-    let range: vscode.Range;
-    let identation: string;
+    sourceCode = removeRegions(sourceCode);
 
-    if (editor)
-    {
-        sourceCode = editor.document.getText();
-        sourceCode = removeRegions(sourceCode);
+    let identation = getIdentation(sourceCode);
+
+    // organize type aliases, interfaces, classes, enums, functions and variables
+    let sourceFile = ts.createSourceFile(fileName, sourceCode, ts.ScriptTarget.Latest, false, ts.ScriptKind.TS);
 
-        identation = getIdentation(sourceCode);
+    let elements = new Transformer().analyzeSyntaxTree(sourceFile);
 
-        // organize type aliases, interfaces, classes, enums, functions and variables
-        sourceFile = ts.createSourceFile(editor.document.fileName, sourceCode, ts.ScriptTarget.Latest, false, ts.ScriptKind.TS);
+    if (elements.filter(x => !(x instanceof UnknownNode)))
```

Comment:
@aljazsim this line is from original code, but it has no sense - `.filter` returns a list anyway, so it's always `true`

maybe it should be `.find`? or `.filter().length` ?